src/Microsoft.Tye.Core/ProcessUtil.cs

```diff
@@ -118,7 +120,12 @@ public static class ProcessUtil
                 {
                     // Even though the Exited event has been raised, WaitForExit() must still be called to ensure the output buffers
                     // have been flushed before the process is considered completely done.
-                    process.WaitForExit();
+                    // Because of the bug in the dotnet runtime https://github.com/dotnet/runtime/issues/29232, Process.WaitForExit()
+                    // hangs for processes that spawn another long-running processes.
+                    // Since these are expected to be long running processes and we're typically not concerned with capturing all of its output
+                    // i.e. it's probably ok for some output to be lost on shutdown, since Tye is shutting down anyway,
+                    // we call Process.WaitForProcessExit(ProcessExitTimeoutMs).
+                    process.WaitForExit(ProcessExitTimeoutMs);
                 }
 
                 if (throwOnError && process.ExitCode != 0)
```

Comment:
If a timeout occurs, what is the `process.ExitCode`? Should we explicitly check for timeout and do something different when it occurs?

Reply:
This fix is a workaround for a bug in dotnet-runtime - [dotnet/runtime#29232](https://github.com/dotnet/runtime/issues/29232) where the child process is shut down, the call `WaitForExit()` never returns.
This bug appeared just now because we switched our dapr support to use `dapr.exe` which in turn calls a long-running process `daprd` as opposed to calling `daprd` directly.

Now, checking for the exit code in case of a timeout would make sense to probably show a warning to the user that the process didn't shut down gracefully, but at this point, there is no way for us to know whether the child process *really* didn't shut down gracefully or it's just because of the bug in dotnet runtime.

Therefore I think it wouldn't add much value to check for the exit code and do additional actions in the case of timeout.